### PR TITLE
Add DeleteFolder operation

### PR DESCRIPTION
**Pull request by hellsworth**

### CODE REVIEW

File: src/types/delete_folder.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    /// The status of the corresponding request, i.e. whether it succeeded or
+    /// resulted in an error.
+    #[serde(rename = "@ResponseClass")]
+    pub response_class: ResponseClass,
+
+    pub response_code: Option<ResponseCode>,
+
+    pub message_text: Option<String>,
+
+    pub folder_ids: Vec<BaseFolderId>,
```
Comment (hellsworth) [inline src/types/delete_folder.rs:76]: I'm not sure what to put here if I can't Deserialize on a BaseFolderID.

Comment (hellsworth) [reply src/types/delete_folder.rs]: From [DeleteFolderResponseMessage](https://learn.microsoft.com/en-us/exchange/client-developer/web-service-reference/deletefolderresponsemessage), there's nothing additional I need besides the response class, response code, and message text.

**Review by babolivier — APPROVED** [review]
Looks good to me! According to Heather, this has been tested in Thunderbird and works, so I think it should be good to merge now.

File: src/types/common.rs [hunk 2]
```diff
@@ -769,6 +769,17 @@ pub struct FolderId {
     pub change_key: Option<String>,
 }
 
+/// The element that determines how elements are deleted.
```
Comment (leftmostcat) [inline src/types/common.rs:772]:
```suggestion
/// The manner in which items or folders are deleted.
```

suggestion: This is a little bit more explicit about what it affects.

File: src/types/delete_folder.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+use serde::Deserialize;
+use xml_struct::XmlSerialize;
+
+use crate::{
+    types::sealed::EnvelopeBodyContents, BaseFolderId, DeleteType, Operation, OperationResponse,
+    ResponseClass, ResponseCode, MESSAGES_NS_URI,
+};
+
+/// A request to delete a folder.
```
Comment (leftmostcat) [inline src/types/delete_folder.rs:13]:
```suggestion
/// A request to delete one or more folders.
```

suggestion: The request can specify multiple folders, so we should reflect that in comments.

File: src/types/delete_folder.rs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+///
+/// See <https://learn.microsoft.com/en-us/exchange/client-developer/web-service-reference/deletefolder>
+#[derive(Clone, Debug, XmlSerialize)]
+#[xml_struct(default_ns = MESSAGES_NS_URI)]
+pub struct DeleteFolder {
+    /// DeleteFolder uses the DeleteType attribute
```
Comment (leftmostcat) [inline src/types/delete_folder.rs:19]:
```suggestion
    /// The method the EWS server will use to perform the deletion.
```

suggestion: Both the context in code and in generated documentation give us the context for determining types, but it's harder to tell what effect setting the field has. This wording is copied directly from `DeleteItem`. Additionally, doc comments should conventionally end with a period (with the exception of the links to external documentation).

File: src/types/delete_folder.rs [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    ///
+    /// See <https://learn.microsoft.com/en-us/exchange/client-developer/web-service-reference/deleteitem#deletetype-attribute>
+    #[xml_struct(attribute)]
+    pub delete_type: DeleteType,
+
+    /// DeleteFolder requires FolderIds which is an array of FolderId and DistinguishedFolderId elements
```
Comment (leftmostcat) [inline src/types/delete_folder.rs:25]:
```suggestion
    /// A list of folders to delete.
```

suggestion: Descriptive of the purpose/effect.

File: src/types/delete_folder.rs [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+/// A request to delete a folder.
+///
+/// See <https://learn.microsoft.com/en-us/exchange/client-developer/web-service-reference/deletefolder>
+#[derive(Clone, Debug, XmlSerialize)]
+#[xml_struct(default_ns = MESSAGES_NS_URI)]
+pub struct DeleteFolder {
+    /// DeleteFolder uses the DeleteType attribute
+    ///
+    /// See <https://learn.microsoft.com/en-us/exchange/client-developer/web-service-reference/deleteitem#deletetype-attribute>
```
Comment (leftmostcat) [inline src/types/delete_folder.rs:21]:
```suggestion
    /// See <https://learn.microsoft.com/en-us/exchange/client-developer/web-service-reference/deletefolder#deletetype-attribute>
```

nitpick: The information should be the same, so this is a quibble more than anything, but the `DeleteFolder` docs should point to the `DeleteFolder` page.

**Review by leftmostcat — APPROVED** [review]
Looks good, thank you! I have a couple of suggestions for improving comments, but nothing that would block merging.

**Pull request merged**